site/src/pages/WorkspacePage/WorkspaceActions/Buttons.tsx

```diff
@@ -190,3 +192,24 @@ export const RetryButton: FC<RetryButtonProps> = ({
     </TopbarButton>
   );
 };
+
+interface FavoriteButtonProps {
+  handleAction: (workspaceID: string) => void;
+  workspaceID: string;
+  isFavorite: boolean;
+}
+
+export const FavoriteButton: FC<FavoriteButtonProps> = ({
+  handleAction,
```

Comment:
It's an implicit convention, but when dealing with event handlers, we usually name them using the `on` prefix like `onSomething` instead of `handleSomething`. So instead of `handleActions`, I think a better name would be `onToggle`.

Also, you can move the query inside this button 😁. We're trying to put things close to where they're used.

Reply:
> Also, you can move the query inside this button 😁. We're trying to put things close to where they're used.

I'm not quite sure how to accomplish this; React doesn't seem to like me putting `useMutation` inside a component.

Reply:
Created https://github.com/coder/coder/issues/11892 to follow-up.